graphite-clickhouse.go

```diff
@@ -108,15 +400,46 @@ func main() {
 		"Additional pprof listen addr for non-server modes (tagger, etc..), overrides pprof-listen from common ",
 	)
 
-	sdList := flag.Bool("sd-list", false, "List registered nodes in SD")
-	sdDelete := flag.Bool("sd-delete", false, "Delete registered nodes for this hostname in SD")
-	sdEvict := flag.String("sd-evict", "", "Delete registered nodes for  hostname in SD")
-	sdClean := flag.Bool("sd-clean", false, "Cleanup expired registered nodes in SD")
-	sdExpired := flag.Bool("sd-expired", false, "List expired registered nodes in SD")
-
 	printVersion := flag.Bool("version", false, "Print version")
 	verbose := flag.Bool("verbose", false, "Verbose (print config on startup)")
 
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
+
+		flag.PrintDefaults()
+
+		fmt.Fprintf(os.Stderr, "\n\nAdditional commands:\n")
+		fmt.Fprintf(os.Stderr, "	-sd-list	List registered nodes in SD\n")
+		fmt.Fprintf(os.Stderr, "	-sd-delete	Delete registered nodes for local hostname in SD\n")
+		fmt.Fprintf(os.Stderr, "	-sd-evict	Delete registered nodes for  hostnames in SD\n")
+		fmt.Fprintf(os.Stderr, "	-sd-clean	Cleanup expired registered nodes in SD\n")
+		fmt.Fprintf(os.Stderr, "	-sd-expired	List expired registered nodes in SD\n")
+		fmt.Fprintf(os.Stderr, "	-match	Match metric against rollup rules\n")
+	}
+
+	if len(os.Args) > 0 {
+		switch os.Args[1] {
+		case "-sd-list":
```

Comment:
I think it is better to not prefix subcommands with '-' to easily distinguish them from flags.

Reply:
This for compatibility with old command line. But not critical and can be changed.